### PR TITLE
Fixes bugs, improvements, tests in stake delegation

### LINKED ISSUE #104: Treasury – Staking Delegation
# Background
We have already built an MVP for our Treasury Dashboard with basic functions (Dashboard, Payments, Settings). This ticket focuses on adding a "Stake Delegation" feature specifically for managing the staking of liquid tokens available within the DAO. A separate feature for handling locked tokens will be introduced in a later ticket. 

# User Stories
* As a DAO member with "create" permission, I want to be able to create staking and unstaking requests so that I can manage the DAO's token delegation and earn rewards.
 * As a DAO member with "vote" permission, I want to be able to review and approve/reject staking and unstaking requests so that I can participate in the DAO's treasury governance process.
 * As a general user, I want to be able to view the history of staking and unstaking requests so that I can track the DAO's delegation activities.

# Acceptance Criteria

## Stake Delegation Page

- [x] Add a "Stake Delegation" link to the Treasury Dashboard navigation.
- [x] Clicking the link displays a page with two tabs: Pending Requests and History. The number of requests in pending history is shown next to the titles.
- [x] Users with create permission can see and click on the Create Request button dropdown, which has two options: Staking, Unstaking.
- [x] There is a filter tab that allows user to enable/disable and rearrange columns within the table below
- [x] There is pagination at the bottom to handle a large number of requests (e.g. greater than 20).

### Pending Requests Tabs

- [x] Displays a list of all pending staking and unstaking requests, including the following columns:
  - [x] ID: The ID of the pending request (default sort, ascending order from 01)
  - [x] Created Date: The date and time the request was made (in UTC)
  - [x] Amount: In NEAR and USD equivalence
  - [x] Validator: Include chosen validator URL, fee, and status
  - [x] Created By: The username of the admin who created the request
  - [x] Notes: Any notes entered by request creator
  - [x] Votes: Number of votes required for approval and current number of completed votes
  - [x] Approvers: The usernames of the users who approved or rejected and their vote indication
  - [x] Actions: Approve, Reject – Only visible to users with vote permissions
  - [x] Details – Shows transaction details
  - [ ] Pencil edit (not shown in diagram) – this button allows users with create permission to edit the request before it had any votes. Clicking on this will show the create staking or unstaking dialogue with the pre-populated information, along with a delete button option.

### Create Staking Request
- [ ] User with create permission who click on create staking request will see the dialogue with the following elements:
  - [x] Display treasury's available balance and total staked amount.
  - [x] Allow users to select a validator from a dynamically fetched list (need to clarify where this is coming from), showing the validator's account, fee, and status.
  - [x] Amount to stake input, with a warning displayed if user tries to exceed the available balance.
  - [x] Notes text area for optional comments.
  - [ ] Delete button (show this to users with create permissions only when editing an existing staking request, not during the initial creation).
  - [x] Cancel button – will clear all fields and close the dialogue.
  - [x] Submit button – triggers a confirmation screen and will then make the request appear in the "Pending Requests" table.

### Create Unstaking Request

- [x] User with create permission who click on create unsticking request will see the dialogue with the following elements:
  - [x] Display treasury's available balance and total staked amount.
  - [ ] Only show a list of selected validators with active staking for the treasury, including the validator's account, fee, status, and staked amount (NEAR and USD equivalence).
  - [x] Amount to unstake, with a warning displayed if user tries to exceed the staking balance.
  - [x] Notes text area for optional comments.
  - [ ] Delete button (show this to users with create permissions only when editing an existing staking request, not during the initial creation).
  - [x] Cancel button – will clear all fields and close the dialogue.
  - [x] Submit button – triggers a confirmation screen and will then make the request appear in the "Pending Requests" table.

### History Tab

- [x] Display a list of all past staking and unstaking requests, with their status (Approved, Expired, Rejected).
- [ ] Include the same columns as the "Pending Requests" tabs, with the addition of a "Transaction Date" and "Transaction Link" columns.

# Request Execution
- [x] Requests are executed (tokens are staked or unstated) only after the required number of approval is met, as defined in the voting policy settings
- [x] Once executed, the request moves to the "History" tab"

# Permission Considerations
- [x] All users can view the "Stake Delegation" page and both tabs.
- [x] Users with "create" permission can create staking and unstaking requests.
- [ ] Users with "create" permission can delete requests before they were approved. These requests will not show up in history.
- [x] Users with "vote" permission can approve or reject pending requests.

# Attachments

Ask @FREZZZBE for the latest mockups.

# Pending Requests Tab
Image

## Create Staking Request
Image

## Create Unstaking Request
Image

## Create Unstaking Request – No Validators
Image

# History Tab
Image

### LINKED ISSUE #112: Tests for stake delegation
- [ ] Should view pending and history requests
- [ ] Should create stake delegation request, should throw error when invalid data is provided
- [ ] Should show warning when account has no staked tokens
- [ ] Should create unstake delegation request, should throw error when invalid data is provided
- [ ] Should vote on stake request
- [ ] Should throw error with vote when account has insufficient balance

### PR DESCRIPTION
**Pull request by Megha-Dev-19**
Resolves #104 
Resolves #112 
[video.webm](https://github.com/user-attachments/assets/4f56d9f3-eac5-43bb-84da-0737c11029b6)
[video.webm](https://github.com/user-attachments/assets/7ff1a433-fddd-4730-a917-bd006ee83e58)
[video.webm](https://github.com/user-attachments/assets/41dbf6a7-d5fc-4352-91b6-ff38e2eb1662)
[video.webm](https://github.com/user-attachments/assets/a0990843-bc65-4d7b-83c6-e1a4ca8f5012)
[video.webm](https://github.com/user-attachments/assets/d532031d-069e-4b21-8cfd-9f9c0bc8c486)
[video.webm](https://github.com/user-attachments/assets/f0336f06-608f-48f6-9815-d74e9e65520f)
[video.webm](https://github.com/user-attachments/assets/cd83eb2f-124e-4528-9f30-57d053262997)

### CODE REVIEW

**Comment by Megha-Dev-19** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@Megha-Dev-19 Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Do you want to apply for monthly streak? Get 8+ score for a single PR this month and receive boost for race-of-sloths!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/Megha-Dev-19?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/111&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/Megha-Dev-19?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/111&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/Megha-Dev-19)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/Megha-Dev-19)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by petersalomonsen** [comment]
@race-of-sloths score 13

**Review by petersalomonsen — APPROVED** [review]

**Pull request merged**
